src/java/org/apache/cassandra/io/util/LimitingRebufferer.java

```diff
@@ -31,29 +31,23 @@
  * Instantiated once per RandomAccessReader, thread-unsafe.
  * The instances reuse themselves as the BufferHolder to avoid having to return a new object for each rebuffer call.
```

Comment:
Let's add "Only one `BufferHolder` can be active at any time, i.e. it is an error to call rebuffer before the previous buffer is released." and copy these lines to `WrappingRebufferer`, `TailOverridingRebufferer` and `BufferManagingRebufferer`.